lib/puppet/provider/foreman_subnet/rest.rb

```diff
@@ -0,0 +1,263 @@
+Puppet::Type.type(:foreman_subnet).provide(:rest) do
+
+  confine :feature => :foreman_api
+
+  def subnets
+    ForemanApi::Resources::Subnet.new({
+      :base_url => resource[:base_url],
+      :oauth    => {
+        :consumer_key    => resource[:consumer_key],
+        :consumer_secret => resource[:consumer_secret]
+      }
+    }, {
+      :headers => {
+        :foreman_user => resource[:effective_user]
+      }
+    })
+  end
+
+  def subnet
+    if @subnet
+      @subnet
+    else
+      spi = subnets.index[0]
```

Comment:
nitpick that we ought to solve in the proxy provider too, but we could use a search query (name = foo) rather than pulling all results and iterating through them